modules/host/info.go

```diff
@@ -57,21 +104,24 @@ func (h *Host) RPCMetrics() modules.HostRPCMetrics {
 func (h *Host) SetSettings(settings modules.HostSettings) error {
 	h.mu.Lock()
 	defer h.mu.Unlock()
+	// Resource lock is grabbed for the save function.
 	h.resourceLock.RLock()
 	defer h.resourceLock.RUnlock()
 	if h.closed {
 		return errHostClosed
 	}
 
-	// Check that the unlock hash was not changed.
+	// Check that none of the illegal fields have been modified.
```

Comment:
perhaps the structs should be split up. It sounds like they have diverged significantly. It would be convenient if one could be embedded in the other, but perhaps not even that is possible.